travis/test-appimages.sh

```diff
@@ -14,12 +14,49 @@ error() {
 
 # first of all, try to run appimagetool
 out/appimagetool-"$ARCH".AppImage && error  # should fail due to missing parameter
+out/appimaged-"$ARCH".AppImage && error  # should fail due to missing parameter
 out/appimagetool-"$ARCH".AppImage -h || error  # should not fail
+out/appimaged-"$ARCH".AppImage -h || error  # should not fail
 
-# print version and update information
+# print versions, updateinfo and other info, if present
+# first, appimagetool
 out/appimagetool-"$ARCH".AppImage --version || error  # should not fail
 out/appimagetool-"$ARCH".AppImage --appimage-version || error  # should not fail
 out/appimagetool-"$ARCH".AppImage --appimage-updateinformation || error  # should not fail
+out/appimagetool-"$ARCH".AppImage --appimage-updateinfo || error  # should not fail
+out/appimagetool-"$ARCH".AppImage --appimage-signature || error  # should not fail
+out/appimagetool-"$ARCH".AppImage --appimage-offset || error  # should not fail
+# next, appimaged
+out/appimaged-"$ARCH".AppImage --version || error  # should not fail
+out/appimaged-"$ARCH".AppImage --appimage-version || error  # should not fail
+out/appimaged-"$ARCH".AppImage --appimage-updateinformation || error  # should not fail
+out/appimaged-"$ARCH".AppImage --appimage-updateinfo || error  # should not fail
+out/appimaged-"$ARCH".AppImage --appimage-signature || error  # should not fail
+out/appimaged-"$ARCH".AppImage --appimage-offset || error  # should not fail
+
+# check creation of .home and .config files (even if never used by these tools)
+out/appimagetool-"$ARCH".AppImage --appimage-portable-home || error  # should not fail
+out/appimagetool-"$ARCH".AppImage --appimage-portable-config || error  # should not fail
+out/appimaged-"$ARCH".AppImage --appimage-portable-home || error  # should not fail
+out/appimaged-"$ARCH".AppImage --appimage-portable-config || error  # should not fail
+[ -d out/appimagetool-"$ARCH".AppImage.home ] || error   # should not fail
+[ -d out/appimagetool-"$ARCH".AppImage.config ] || error   # should not fail
+[ -d out/appimaged-"$ARCH".AppImage.home ] || error   # should not fail
+[ -d out/appimaged-"$ARCH".AppImage.config ] || error   # should not fail
+# Don't care cleaning up out/appimage{d,tool}-"$ARCH".AppImage.{config,home}... Or should we?
+
+# now test --appimage-mount
+timeout "$TIMEOUT" out/appimaged-"$ARCH".AppImage    --appimage-mount ; ret_of_appimaged_mount="$?"
+timeout "$TIMEOUT" out/appimagetool-"$ARCH".AppImage --appimage-mount ; ret_of_appimagetool_mount="$?"
+echo ret_of_appimaged_mount=$ret_of_appimaged_mount
+echo ret_of_appimagetool_mount=$ret_of_appimagetool_mount
+
+if [[ x$ret_of_appimaged_mount != x124 ]] || [[ x$ret_of_appimagetool_mount != x124 ]] ; then
+    echo "Error: mounting of appimaged or appimagetool did not terminate by timeout as expected" >&2
```

Comment:
See above. Also, just for your information, you could use `"$ret_of_appimaged_mount" == ""`. Beware that both solutions are bash-only (not a problem for us, as the scripts are forced to run with bash, but in general one needs to be careful about "bashisms").

Reply:
I know `[[ ... ]]` is a bashism. But I don't know to write for `/bin/sh` anyway, and never was ambitious (or forced to) to do so.

Reply:
@KurtPfeifle the string comparison is supported in the old-style `[ ... ]` bash comparisons, too. What makes it a bashism is that the original Bourne shell doesn't support string comparisons at all.

I was just trying to share some of my knowledge, to prevent you from running into issues on Debian etc., where bash isn't always installed by default.